packages/qwik-city/src/middleware/request-handler/request-event.ts

```diff
@@ -220,6 +244,26 @@ export function createRequestEvent(
       return new RedirectMessage();
     },
 
+    rewrite: (pathname: string) => {
+      check();
+      if (pathname.startsWith('http')) {
+        throw new Error('Rewrite does not support absolute urls');
+      }
+
+      const rewriteUrl = new URL(url);
+      rewriteUrl.pathname = pathname;
+
+      // Fix consecutive slashes - e.g //path//to//page -> /path/to/page
+      const fixedPathname = rewriteUrl.pathname.replace(/(^|[^:])\/{2,}/g, '$1/');
+      if (rewriteUrl.pathname !== fixedPathname) {
+        console.warn(`Rewrite URL ${rewriteUrl.pathname} is invalid, fixing to ${fixedPathname}`);
```

Comment:
there's no point to this warning, you always need to do the .replace anyway so even if you fix it in the code the runtime stays the same. You can replace during the assignment.

Also, the replace can just be `pathname.replace(/\/+/g, '/')`, no? The pathname shouldn't include the protocol?